src/v/pandaproxy/schema_registry/storage.h

```diff
@@ -143,4 +144,17 @@ class schema_key_handler : public json::base_handler<Encoding> {
     }
 };
 
+inline schema_key schema_key_from_iobuf(iobuf&& iobuf) {
+    auto p = iobuf_parser(std::move(iobuf));
+    auto str = p.read_string(p.bytes_left());
+    return json::rjson_parse(str.data(), schema_key_handler<>{});
```

Comment:
> depending on how this serialization/deserialization is used, you might consider using our internal serde format which encodes the string with its length instead of relying the iobuf size to implicitly encode the size.
> 
> ```
> iobuf = reflection::serialize(iobuf, string);
> ...
> auto p = iobuf_parser(std::move(iobuf));
> auto string = reflection::adl<ss::sstring>{}.from(p);
> ```

This requires compatibility, so I don't have much freedom here. I guess versioning is, or is planned to done with the `magic` field.

Reply:
got it. a strategic comment that reminds readers that you're explicitly trying to maintain compatibility with a 3rdparty format would be useful.